executor/infoschema_reader.go

```diff
@@ -565,7 +567,27 @@ func (e *hugeMemTableRetriever) setDataForColumns(ctx sessionctx.Context) error
 	return nil
 }
 
-func (e *hugeMemTableRetriever) dataForColumnsInTable(schema *model.DBInfo, tbl *model.TableInfo) {
+func (e *hugeMemTableRetriever) dataForColumnsInTable(ctx context.Context, sctx sessionctx.Context, schema *model.DBInfo, tbl *model.TableInfo) {
+	if tbl.IsView() {
+		// Retrieve view columns info.
+		planBuilder, _ := plannercore.NewPlanBuilder(sctx, infoschema.GetInfoSchema(sctx), &hint.BlockHintProcessor{})
```

Comment:
Can we use a function to handle this logic so that we can reuse the code?

Reply:
Yes. But we should discuss which place this function should place, it will be called in "executor/show.go" and "executor/infoschema_reader.go". I think "infoschema/infoschema.go" is a good place for it, or we just put it to "executor/show.go" or "executor/infoschema_reader.go". What do you think?

Reply:
Updated. I test on my side, place it to "infoschema/infoschema.go" is not work, it will cause cycle import.